backend/src-common/src/main/java/org/eclipse/sw360/datahandler/db/BulkDeleteUtil.java

```diff
@@ -0,0 +1,773 @@
+        this.componentRepository = componentRepository;
+        this.releaseRepository = releaseRepository;
+        this.projectRepository = projectRepository;
+        this.componentModerator = componentModerator;
+        this.releaseModerator = releaseModerator;
+        this.attachmentConnector = attachmentConnector; 
+        this.attachmentDatabaseHandler = attachmentDatabaseHandler;
+        this.dbHandlerUtil = dbHandlerUtil;
+        this.inspector = null;
+    }
+
+    public BulkOperationNode deleteBulkRelease(String releaseId, User user, boolean isPreview) throws SW360Exception  {
+        if (!BackendUtils.IS_BULK_RELEASE_DELETING_ENABLED) {
+            throw fail(500, "Bulk release deleting feature is not enabled.");
+        }
+        if (!PermissionUtils.isAdmin(user)) {
+            throw fail(403, "Failed to check the admin privilege.");
+        }
+        if (!PermissionUtils.IS_ADMIN_PRIVATE_ACCESS_ENABLED) {
+            throw fail(500, "Admin private access is not enabled.");
+        }
+        
+        Release release = releaseRepository.get(releaseId);
+        assertNotNull(release, "No releases found to bulk delete!");
+        
+        //create a result state object
+        Map<String, BulkOperationResultState> resultStateMap = new HashMap<String, BulkOperationResultState>();
+        
+        //create an all linked release map
+        Map<String, Release> allLinkedReleaseMap = new HashMap<String, Release>();  
+        getAllLinkedReleaseMap(release.getId(), allLinkedReleaseMap);
+        Set<String> allLinkedReleaseIds = allLinkedReleaseMap.keySet();
+
+        //create a deep copied release map for temporary data.
+        Map<String, Release> workReleaseMap = getDuplicatedReleaseMap(allLinkedReleaseMap);
+
+        //create an external flag map and an ids map of referencing releases
+        Map<String, Boolean> externalLinkMap = new HashMap<String, Boolean>();
+        Map<String, List<String>> referencingReleaseIdsMap = new HashMap<String, List<String>>();
+        getExternalLinkMap(allLinkedReleaseMap.keySet(), externalLinkMap, referencingReleaseIdsMap);
+        
+        //update result state map for excluded releases
+        for (String externalLinkMapId : externalLinkMap.keySet()) {
+            if (externalLinkMap.get(externalLinkMapId)) {
+                resultStateMap.put(externalLinkMapId, BulkOperationResultState.EXCLUDED);
+            }
+        }
+        
+        //create a linked component map
+        Map<String, Component> allComponentMap = getComponenMap(allLinkedReleaseMap);
+        Map<String, Component> workComponentMap = getDuplicatedComponentMap(allComponentMap);
+        
+        if (inspector != null) inspector.checkVariables(workReleaseMap, workComponentMap, externalLinkMap, referencingReleaseIdsMap);
+        
+        //delete linked releases
+        boolean isCompleted = false;
+        for (int i=0; i<LOOP_MAX; i++) {
```

Comment:
Use proper variable name for i.

Reply:
Done. I use loopCount.
Note : In other SW360 codes, 'i' is often used. This is certainly not a good idea.